### PR TITLE
School Library

### PR DESCRIPTION
**Pull request by altontonn**
# OOP school library: add Person, Student, and Teacher classes
In this PR, I added three files for a person, student and teacher

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Student < Person
+  def initialize(name, age, parent_permission, classroom)
+    super(age, name, parent_permission)
```
Comment (ShadyShawkat) [inline student.rb]: - [ ] Kindly make sure that when passing in the parameters to the `super` class, the order of the parameters matches with the super constructor. Currently, the super constructor accepts parameters in the following order:
https://github.com/altontonn/School-Library/blob/bd96b9a4fd7b1b3e7958f5c111a3b1e228a5e966/person.rb#L2

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require './person'
+class Teacher < Person
+  def initialize(specialization, name, parent_permission: 'true')
+    super(age, name, parent_permission)
```
Comment (ShadyShawkat) [inline teacher.rb]: - [ ] Kindly make sure that when passing in the parameters to the `super` class, the order of the parameters matches with the super constructor. Currently, the super constructor accepts parameters in the following order:
https://github.com/altontonn/School-Library/blob/bd96b9a4fd7b1b3e7958f5c111a3b1e228a5e966/person.rb#L2

- [ ] Please add the `age` to the constructor, as you're passing it to the super constructor.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @altontonn, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @altontonn  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**